src/utils/TransactionUtils.ts

```diff
@@ -79,23 +79,54 @@ export async function runTransaction(
       // If error is due to a nonce collision or gas underpricement then re-submit to fetch latest params.
       retriesRemaining -= 1;
       logger.debug({
-        at: "TxUtil",
+        at: "TxUtil#runTransaction",
         message: "Retrying txn due to expected error",
         error: JSON.stringify(error),
         retriesRemaining,
       });
 
       return await runTransaction(logger, contract, method, args, value, gasLimit, null, retriesRemaining);
     } else {
-      // If transaction error reason is known to be benign, then reduce the log level to warn.
-      logger[txnRetryable(error) ? "warn" : "error"]({
-        at: "TxUtil",
-        message: "Error executing tx",
-        retriesRemaining,
-        error: JSON.stringify(error),
-        notificationPath: "across-error",
-      });
-      throw error;
+      // Empirically we have observed that Ethers can produce nested errors, so we try to recurse down them
+      // and log them as clearly as possible. For example:
+      // - Top-level (Contract method call): "reason":"cannot estimate gas; transaction may fail or may require manual gas limit" (UNPREDICTABLE_GAS_LIMIT)
+      // - Mid-level (eth_estimateGas): "reason":"execution reverted: delegatecall failed" (UNPREDICTABLE_GAS_LIMIT)
+      // - Bottom-level (JSON-RPC/HTTP): "reason":"processing response error" (SERVER_ERROR)
+      if (typeguards.isEthersError(error)) {
+        const ethersErrors: { reason: string; err: EthersError }[] = [];
+        let topError: EthersError = error as EthersError;
+        while (topError !== undefined) {
+          ethersErrors.push({ reason: topError.reason, err: topError.error });
+          topError = topError.error;
+        }
+        logger[ethersErrors.some((e) => txnRetryable(e.err)) ? "warn" : "error"]({
+          at: "TxUtil#runTransaction",
+          message: "Error executing tx",
+          retriesRemaining,
+          errorReasons: ethersErrors.map((e, i) => `\n\t ${i}: ${e.reason}`),
```

Comment:
Should we join these errors?

```suggestion
          errorReasons: ethersErrors.map((e, i) => `\t ${i}: ${e.reason}`).join('\n'),
```